sktime/forecasting/compose/tests/test_multiplex.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Tests for MultiplexForecaster and associated dunders."""
+
+__author__ = ["aiwalter", "miraep8"]
+
+from sktime.datasets import load_shampoo_sales
+from sktime.forecasting.all import (
+    AutoETS,
+    ExpandingWindowSplitter,
+    ForecastingGridSearchCV,
+    MultiplexForecaster,
+    NaiveForecaster,
+    ThetaForecaster,
+)
+from sktime.forecasting.model_evaluation import evaluate
+from sktime.utils.validation.forecasting import check_scoring
+
+
+def _score_forecasters(forecasters, cv, y):
+    """Will evaluate all the forecasters on y and return the name of best."""
+    scoring = check_scoring(None)
+    scoring_name = f"test_{scoring.name}"
+    score = None
+    for name, forecaster in forecasters:
+        results = evaluate(forecaster, cv, y)
+        results = results.mean()
+        new_score = float(results[scoring_name])
+        if not score or new_score < score:
+            score = new_score
+            best_name = name
+    return best_name
+
+
+def test_multiplex_forecaster_alone():
+    """Test results of MultiplexForecaster.
+
+    Because MultiplexForecaster is in many ways a wrapper for an underlying
+    forecaster - we can confirm that if the selected_forecaster is set that the
+    MultiplexForecaster performs as expected.
+    """
+    from numpy.testing import assert_array_equal
+
+    y = load_shampoo_sales()
+    # Note - we select two forecasters which are deterministic.
+    forecaster_tuples = [
+        ("naive", NaiveForecaster()),
+        ("theta", ThetaForecaster()),
+    ]
+    forecaster_names = [name for name, _ in forecaster_tuples]
```

Comment:
I think we need to be careful, there is a subtle point here:
objects are being passed by reference in python. So, both the `MultiplexForecaster` hyper-parameters and the `forecasters` list are pointing to the *identical* forecaster objects!
They should be cloned inside the `MultiplexForecaster`, but you could think up a situation where two bugs cancel each other out - i.e., if clone is not done properly.
I would recommend a deepcopy to have the two sides of the comparison, or separate construction from the class.

Reply:
Oof, my bad!  Very important point! Will be sure to fix that immediately, thanks for spotting!

Reply:
Oh actually - I do remember this! I think I was being a bit lazy as I noticed that MultiplexForecaster does technically make a deep copy via sklearn.base.clone (see below)  But still, this code is ugly regardless and I will fix it. 

  ```
  def _set_forecaster(self):
        self._check_selected_forecaster()
        if self.selected_forecaster is not None:
            for name, forecaster in self.forecasters:
                if self.selected_forecaster == name:
                    self.forecaster_ = clone(forecaster)
```

Reply:
> as I noticed that MultiplexForecaster does technically make a deep copy via sklearn.base.clone (see below)

Yes, I know - my point being, if that gets broken somehow (and it is not difficult to break this, with all the underscores or not-underscores floating around), then the test will no longer pick up if the tested property fails...

Reply:
Ah, got it!  Sorry, and yes I totally agree.  Better to have code that is easier to break.  Thanks! Hopefully this resolves it for you!